src/DotNetCore.CAP.Kafka/CAP.KafkaOptions.cs

```diff
@@ -48,6 +48,11 @@ public KafkaOptions()
         /// </summary>
         public Func<ConsumeResult<string, byte[]>, List<KeyValuePair<string, string>>>? CustomHeaders { get; set; }
 
+        /// <summary>
+        /// If you need to get offset and partition and so on.., you can use this function to write additional header into <see cref="CapHeader"/>
+        /// </summary>
+        public Func<ConsumeResult<string, byte[]>,IServiceProvider, List<KeyValuePair<string, string>>>? CustomHeadersBuilder { get; set; }
```

Comment:
I suggest Add `[Obsolete("xxxxx")]` for old `CustomHeaders` option, we will  remove old option at the feature